rclpy/rclpy/node.py

```diff
@@ -207,11 +207,7 @@ def __init__(
             self._parameter_overrides.update({p.name: p for p in parameter_overrides})
 
         # Clock that has support for ROS time.
-        # Note: parameter overrides and parameter event publisher need to be ready at this point
-        # to be able to declare 'use_sim_time' if it was not declared yet.
         self._clock = ROSClock()
-        self._time_source = TimeSource(node=self)
-        self._time_source.attach_clock(self._clock)
 
         if automatically_declare_parameters_from_overrides:
             self.declare_parameters(
```

Comment:
I believe that `ROSClock()` is required to prior to `declare_parameters` for sure.

`ParameterEvetns` from this call will be based on system timestamp, even if it overrides `use_sim_time` parameter into true, because TimeSource is not attached yet.